sphinx_gallery/tests/test_gen_gallery.py

```diff
@@ -393,5 +393,16 @@ def test_first_notebook_cell_config(sphinx_app_wrapper):
         parse_config(sphinx_app_wrapper.create_sphinx_app())
 
 
+@pytest.mark.conf_file(content="""
+sphinx_gallery_conf = {
+    'backreferences_dir': False,
+}""")
+def test_backreferences_dir_config(sphinx_app_wrapper):
+    """Tests 'backreferences_dir' type checking."""
+    from sphinx_gallery.gen_gallery import parse_config
+    with pytest.raises(ValueError):
```

Comment:
Better to always do `pytest.raises(ValueError, match='some substring'):` as it ensures you're catching the right thing, and is more explicit about it, too

Reply:
Is this adequate or should I put more of the message for `match`?

Reply:
Looks great